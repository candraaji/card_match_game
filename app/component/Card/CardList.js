import React from 'react';
import Card from './Card';

const CardList = React.memo(function CardList({
  onPressCard,
  listCard,
  countOpenCard,
  setStepToOpen,
}) {
  return listCard.map((dataCard, index) => (
    <Card
      dataCard={dataCard}
      setStepToOpen={setStepToOpen}
      countOpenCard={countOpenCard}
      key={index}
      onPressCard={onPressCard}
    />
  ));
});

export default CardList;
