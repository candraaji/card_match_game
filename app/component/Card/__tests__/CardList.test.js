import React from 'react';
import renderer from 'react-test-renderer';
import CardList from '../CardList';

describe('List Card Component', () => {
  const dataCard = [
    {index: 1, number: 10, status: 'editable', position: 'hide'},
    {index: 2, number: 11, status: 'uneditable', position: 'show'},
    {index: 3, number: 11, status: 'uneditable', position: 'show'},
    {index: 4, number: 10, status: 'editable', position: 'hide'},
  ];

  const setStepToOpen = jest.fn();
  const countOpenCard = 1;
  const onPressCard = jest.fn();

  test('it should render list card compoennt correctly', () => {
    jest.useFakeTimers();
    const wrapper = renderer.create(
      <CardList
        onPressCard={onPressCard}
        listCard={dataCard}
        countOpenCard={countOpenCard}
        setStepToOpen={setStepToOpen}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
