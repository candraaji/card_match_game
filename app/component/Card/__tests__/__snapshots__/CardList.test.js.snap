// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Card Component it should render list card compoennt correctly 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="PressableCard"
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 175,
          "justifyContent": "center",
          "padding": 10,
          "transform": Array [
            Object {
              "rotateY": "180deg",
            },
          ],
          "width": 125,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/cardBackground.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 5,
            "borderWidth": 0.5,
            "height": 175,
            "justifyContent": "center",
            "width": 125,
          }
        }
      />
    </View>
  </View>,
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 175,
        "justifyContent": "center",
        "padding": 10,
        "width": 125,
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      11
    </Text>
  </View>,
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 175,
        "justifyContent": "center",
        "padding": 10,
        "width": 125,
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      11
    </Text>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="PressableCard"
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 175,
          "justifyContent": "center",
          "padding": 10,
          "transform": Array [
            Object {
              "rotateY": "180deg",
            },
          ],
          "width": 125,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/cardBackground.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 5,
            "borderWidth": 0.5,
            "height": 175,
            "justifyContent": "center",
            "width": 125,
          }
        }
      />
    </View>
  </View>,
]
`;
